threadpoolctl.py

```diff
@@ -1219,16 +1227,13 @@ def _get_libc(cls):
         """Load the lib-C for unix systems."""
         libc = cls._system_libraries.get("libc")
         if libc is None:
-            libc_name = find_library("c")
-            if libc_name is None:  # pragma: no cover
-                warnings.warn(
-                    "libc not found. The ctypes module in Python"
-                    f" {sys.version_info.major}.{sys.version_info.minor} is maybe"
-                    " too old for this OS.",
-                    RuntimeWarning,
-                )
-                return None
-            libc = ctypes.CDLL(libc_name, mode=_RTLD_NOLOAD)
+            # Remark: If libc is statically linked or if Python is linked against an
+            # alternative implementation of libc like musl, find_library will return
+            # None and CDLL will load the main program itself which should contain the
+            # libc symbols. We still name it libc for convenience.
+            # If the main program does not contain the libc symbols, it's ok because
+            # we check their presence later anyway.
+            libc = ctypes.CDLL(find_library("c"), mode=_RTLD_NOLOAD)
```

Comment:
I don't check it here because this function is not os dependent. Instead we check in the os specific methods and I added an informative warning when the symbols are not found in this PR. That's why I wrote the comment in lines 1234-1235